bottlecap/src/proc/mod.rs

```diff
@@ -58,6 +60,122 @@ fn get_network_data_from_path(path: &str) -> Result<NetworkData, io::Error> {
+        total_system_time_ms: 0.0,
+        total_idle_time_ms: 0.0,
+        individual_cpu_idle_times: HashMap::new(),
+    };
+
+    // SC_CLK_TCK is the system clock frequency in ticks per second
+    // We'll use this to convert CPU times from user HZ to milliseconds
+    let clktck = clock::get_clk_tck()? as f64;
+
+    for line in reader.lines() {
+        let line = line?;
+        let mut values = line.split_whitespace();
+
+        if let Some(label) = values.next() {
+            if label == "cpu" {
+                // Parse CPU times for total user, system, and idle
+                let user: Option<f64> = values.next().and_then(|s| s.parse().ok());
+                values.next(); // skip "nice"
+                let system: Option<f64> = values.next().and_then(|s| s.parse().ok());
+                let idle: Option<f64> = values.next().and_then(|s| s.parse().ok());
+
+                match (user, system, idle) {
+                    (Some(user_val), Some(system_val), Some(idle_val)) => {
+                        cpu_data.total_user_time_ms = (1000.0 * user_val) / clktck;
+                        cpu_data.total_system_time_ms = (1000.0 * system_val) / clktck;
+                        cpu_data.total_idle_time_ms = (1000.0 * idle_val) / clktck;
+                    }
+                    (_, _, _) => {
+                        return Err(io::Error::new(
+                            io::ErrorKind::InvalidData,
+                            "Failed to parse CPU data",
+                        ))
+                    }
+                }
+            } else if label.starts_with("cpu") {
+                // Parse per core (i.e. "cpu0", "cpu1", etc.) idle times
+                // Skip the first three values (user, nice, system) and get the 4th value (idle)
+                let idle: Option<f64> = values.nth(3).and_then(|s| s.parse().ok());
+
+                match idle {
+                    Some(idle_val) => {
+                        cpu_data
+                            .individual_cpu_idle_times
+                            .insert(label.to_string(), (1000.0 * idle_val) / clktck);
+                    }
+                    None => {
+                        return Err(io::Error::new(
+                            io::ErrorKind::InvalidData,
+                            "Failed to parse per-core CPU data",
+                        ))
+                    }
+                }
+            }
+        }
+    }
+
+    if cpu_data.individual_cpu_idle_times.is_empty() {
+        return Err(io::Error::new(
+            io::ErrorKind::NotFound,
+            "Per-core CPU data not found",
+        ));
+    }
+
+    Ok(cpu_data)
+}
+
+pub fn get_uptime() -> Result<f64, io::Error> {
+    get_uptime_from_path(PROC_UPTIME_PATH)
+}
+
+fn get_uptime_from_path(path: &str) -> Result<f64, io::Error> {
+    let file = File::open(path)?;
+    let reader = io::BufReader::new(file);
+
+    if let Some(Ok(line)) = reader.lines().next() {
```

Comment:
not sure we want to have a `Some(Ok(...))` is there a better way to handle this?

Reply:
Yes I'll change it to be more similar the other methods